README.md

```diff
@@ -1,72 +1,430 @@
+    # requirements, conditions are using the AND semantic.
+    - parameter: someParameter
+      value: 100
+    - AnyOf:
+      - parameter: someOtherParameter
+        value: 10
+      - parameter: someOtherParameter2
+        value: 100
+      - AnyOf:
+        - parameter: someOtherParameter3
+          value: 10
+        - parameter: someOtherParameter4
+          value: 100
+```
+
+#### Achievement Rule `replaces`
+
+The `replaces` key allows you to specify an array of other achievement names.
+Should the user receiving the new achievement have one of the other achievements
+specified in here, they are taken away from the user. This is useful for
+multi-stage achievements where the best "gold" achievement replaces the "silver"
+achievement.
+
+#### Achievement Rule `maxAwarded` and `maxAwardedTotal`
+
+These two settings configure how often the achievement can be granted to a user.
+`maxAwarded` specifies how often a user can have the achievement at the same
+time. `maxAwardedTotal` specifies how often a user can be granted the
+achievement over time. Both of them are optional, however, if both are given,
+make sure that `maxAwarded` is less or equal than `maxAwardedTotal`. If both
+`maxAwarded` and `maxAwardedTotal` are not set, they default to 1. If only
+`maxAwarded` is set, `maxAwardedTotal` defaults to positive infinity. If only
+`maxAwardedTotal` is set, `maxAwarded` defaults to `maxAwardedTotal`.
+
+#### Achievement Rule `actions`
+
+You can use the `actions` array to specify actions to be executed whenever the
+achievement is granted. Each action is an object with a single top-level key
+denoting the action type. Currently, only the "xp" action is supported:
+
+```yml
+actions:
+  - xp:
+    # The name of the xp-pool
+    name: myXP
+    # The amount to add. You can also specify a negative amount here
+    # to subtract xp.
+    amount: 1
+```
+
+#### Achievement Rule `hidden`
+
+This boolean describes whether or not the achievement is "public" and returned 
+when the /user endpoint is used. It defaults to `false`. This is useful for 
+achievements which are only used to represent state and should not be visible to
+the user.
+    
+#### Achievement Rule `scope`
 
-1. Make sure you have [NodeJS](https://nodejs.org/) and [npm](https://www.npmjs.com/) installed.
-2. Install your dependencies
+This is not yet implemented. It is thought to cover cases where the achievement
+is granted in the scope of an entity, for example a course or classroom.
+
+## Internals
+
+This project uses [Feathers](http://feathersjs.com) as api framework and MongoDB
+for storing data. Mongoose is used to validate the stored data. 
+
+### Feathers services
+
+The gamification service is built using four Feathers services:
+
+1. Achievements: The achievements service handles achievement data and stores
+granted achievements in the `achivements` collection. Each row consists of the
+`user_id`, the `name` (achievement name), `amount` (how often the user has
```

Comment:
Shouldn't this be `current_amount` and `total_amount`? In other sections, you already talk about `maxAwardedTotal`, so the new values from your PR should also be included here.